keps/sig-apps/0753-sidecarcontainers.md

```diff
@@ -82,6 +84,24 @@ Check these off as they are completed for the Release Team to track. These check
 
 To solve the problem of container lifecycle dependency we can create a new class of container: a "sidecar container" that behaves primarily like a normal container but is handled differently during termination and startup.
 
+## Prerequisites
+
+On June 23 2020, during SIG-node meeting, it was decided that this KEP has a
+prerequisite on the (not yet submitted KEP) kubelet node graceful shutdown.
+
```

Comment:
I think it would be good to explain what this means in terms of this KEP, when do we consider this KEP implementable? We have in the graduation criteria mentions of the shutdown kep being in beta/GA etc. But when can this feature be moved to Alpha? Should we wait till shutdown is in Beta before implementing this or is it fine to add this feature while shutdown is still in Alpha?

Reply:
Well, that is what we haven't talked about in SIG-node meeting and asked in the PR description :-)

I proposed no dependency for alpha, because: (a) we have the feature kubelet shutdown feature coming and we are in sync (have met several times, etc.) and the interaction seems to be quite small (just respect the pod.DeletionGracePeriodSeconds when killing a sidecar would make the features interoperate correctly), (b) given the other point and that alpha being experimental and we plan to continue in sync, it doesn't seems very so risky to me, (c) it will be nice to get this ball rolling.

But I'd really like to know what @derekwaynecarr thinks. I'm, of course, okay adding a dependency on alpha (sidecar alpha can only be done with kubelet shutdown alpha) or more strict dependency on Beta too. As @derekwaynecarr thinks it's better :-)

Reply:
i think when a clear solution is identified for shutdown, this work can proceed.  i do not see a reason why the features cannot go to beta together.

Reply:
Cool. Will take note to maybe change that in the following PRs. Marking this as solved, then! :)